src/ui/MissingTerm/MissingTerms.ts

```diff
@@ -0,0 +1,188 @@
+import { $$, Initialization } from '../../Core';
+import { Dom } from '../../utils/Dom';
+import { analyticsActionCauseList, IAnalyticsIncludeMissingTerm } from '../Analytics/AnalyticsActionListMeta';
+import { IQueryResult } from '../../rest/QueryResult';
+import XRegExp = require('xregexp');
+
+export interface IMissingTermsOptions {
+  caption?: string;
+  clickable?: boolean;
+  includeInBreadcrumb?: boolean;
+  numberOfResults?: number;
+}
+
+/**
+ * This [result template component](https://docs.coveo.com/en/513/#using-result-template-components) renders a list of query terms
+ * that were not matched by its associated result item.
+ */
+export class MissingTerms extends Component {
+  static ID = 'MissingTerms';
+  static options: IMissingTermsOptions = {
+    /**
+     * Whether to allow the end-user to click a missing term to re-inject it in the query as
+     * an exact phrase match (i.e., as an expression between double quote characters).
+     *
+     * **Default:** `true`
+     */
+    clickable: ComponentOptions.buildBooleanOption({ defaultValue: true }),
+    /**
+     * The text to display before missing terms.
+     *
+     * **Default:** The localized string for `Missing`.
+     */
+    caption: ComponentOptions.buildLocalizedStringOption({ defaultValue: 'Missing' }),
+    /**
+     * The maximum number of missing term to be displayed
+     *
+     * **Default:** `5`
+     * **Minimum value:** `1`
+     */
+    numberOfResults: ComponentOptions.buildNumberOption({
+      defaultValue: 5,
+      min: 1
+    })
+  };
+
+  static doExport = () => {
+    exportGlobally({
+      MissingTerms: MissingTerms
+    });
+  };
+
+  /* Used to split terms and phrases. Match characters that can separate words or caracter for chinese, japanese and korean.
+  * Han: Unicode script for chinesse caracter
+  * We only need to import 1 asian script because what is important here is the space between the caracter and any script will contain it
+  */
+  private wordBoundary = '(([\\p{Han}])?([^(\\p{Latin}-)])|^|$)';
+  private termForcedToAppear: Array<string> = [];
+
+  /**
+   * Creates a new `MissingTerms` component instance.
+   * @param element The element on which to instantiate the component.
+   * @param options The configuration options for the component.
+   * @param bindings The bindings required by the component to function normally. If not set, these will be automatically resolved (with a slower execution time).
+   * @param result The query result item to associate the component with.
+   */
+  constructor(
+    public element: HTMLElement,
+    public options?: IMissingTermsOptions,
+    bindings?: IComponentBindings,
+    public result?: IQueryResult
+  ) {
+    super(element, MissingTerms.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, MissingTerms, options);
+    this.result.absentTerms = [...this.result.absentTerms.splice(0, this.options.numberOfResults)];
```

Comment:
After talking with @olamothe, I'll change this to add a `...` element that, when clicked, will show all the missing term